File: shared/order-pricing/money.js
/**
 * Money amounts: round to 2 decimal places (half-up) for display and calculations.
 */

export function roundMoney(value) {
  const n = Number(value)
  if (!Number.isFinite(n)) return 0
  return Math.round((n + Number.EPSILON) * 100) / 100
}

/** Dollar string with exactly 2 decimals, e.g. "12.34" */
export function formatMoney(value) {
  return roundMoney(value).toFixed(2)
}

/** Display with $ prefix, e.g. "$12.34" */
export function formatMoneyDisplay(value) {
  return `$${formatMoney(value)}`
}

File: backend/utils/money.py
"""Round monetary amounts to 2 decimal places (half-up)."""

from decimal import Decimal, ROUND_HALF_UP

TWOPLACES = Decimal('0.01')


def round_money(value) -> Decimal:
    if value is None:
        return Decimal('0.00')
    if isinstance(value, Decimal):
        d = value
    else:
        d = Decimal(str(value))
    return d.quantize(TWOPLACES, rounding=ROUND_HALF_UP)


def round_money_float(value) -> float:
    return float(round_money(value))
